process: add initial tool management (#932)
eclipse-score/score: docs/platform_management_plan/tool_management.rst

   :status: draft
   :safety: ASIL_B
   :tags: platform_management
   :realizes: wp__tlm_plan

Tool Management/ Tool Management Plan
-------------------------------------

This document implements parts of the :need:`wp__platform_mgmt`.

Purpose
+++++++
The purpose of the Tool Management Plan is to guide the identification, evaluation and qualification of
project tools.

S-CORE objectives is to use OSS tools for the generation of all required work products. Thus all
tools which will be used as part of the tool chain for the work product generation must be identified,
documented and evaluated and if applicable, qualified.

Objectives and Scope
++++++++++++++++++++

Tool Management Goals
^^^^^^^^^^^^^^^^^^^^^

* Tools are identified and documented
* Tools are evaluated
* Tools are qualified, if applicable

Approach
++++++++

Identification
^^^^^^^^^^^^^^

Each tool is identified by its name or UID and version.
The tools are stored in one GitHub repository, `tooling repository <https://GitHub.com/eclipse-score/tooling/>`_.

For each tool the :need:`gd_temp__tool_management__verif_rpt_template` shall be used to document
the relevant information.

> :need:`gd_chklst__tool__cr_review` may used to check that all relevant information is considered.

Author: changed, may was there, if it enforced by tool, may not needed?

The configuration of tools is part of the :need:`doc__config_mgt_plan`.

Evaluation
^^^^^^^^^^

The tool confidence level (TCL) must be determined as outlined in the
:need:`doc_concept__tool__process`.

Step-by-Step guide:

1. Determine the tool impact
2. Determine the tool error detection
3. Determine the tool confidence level

**To Step 1**

Based on the found malfunctions or threats judge, if any safety or security requirements may violated.
If this is the case, determine tool impact to "YES" otherwise to "NO".

If tool impact is "NO", continue with 3, otherwise 2.

**To Step 2**

Based on the found malfunctions or threats judge, if these can be all detected or prevented?
Consider also potential safety measures and security controls to increase the detection and
prevention of these, if applicable.
If there is no guarantee to either detect or prevent these, judge tool error detection to "NO",
otherwise to "YES".

**To Step 3**

> If tool impact is judged to "YES" and tool error detection judged to "YES", the tool confidence

Reviewer: I think it should be "and tool error detection judged to "NO""

Author: changed

Specify requirements for the tools purpose and usage in the S-CORE context.
Analyze the malfunctions and threats and specify safety measures and security controls to increase
the prevention and detection of them.
Specify test cases and tests to verify the requirements and generate a test report.
Update the Tool Verification Report accordingly to document the outcomes.

For the generation of requirements, verification and analyses the processes defined in the
:need:`wp__platform_mgmt` shall be used (Requirements Engineering, Verification,
Safety/Security Analyzes).

Tailoring
^^^^^^^^^
Tailoring of safety activities for confidence in the use of software tools:

The following ISO 26262 defined safety requirements are not relevant for the S-CORE SW platform
development, as they are not used:

:need:`std_req__iso26262__support_11471`, :need:`std_req__iso26262__support_11472`,
:need:`std_req__iso26262__support_11473`, :need:`std_req__iso26262__support_11474`,
:need:`std_req__iso26262__support_11481`, :need:`std_req__iso26262__support_11482`,
:need:`std_req__iso26262__support_11483`

Tool List
^^^^^^^^^

.. list-table:: SW Platform tool overview
    :header-rows: 1

>     * - tool Id

Reviewer: This is more a tool list for the "safety and security management". I would like to add here info/links relevant to the user like version, user manual, use case (or at least state that these are part of the "verification report").

Author: add link, but see comment above, how to create tool list

   # SPDX-License-Identifier: Apache-2.0
   # *******************************************************************************

.. document:: Tool Management Plan
   :id: doc__platform_tool_management_plan
   :status: draft
   :safety: ASIL_B
   :tags: platform_management
   :realizes: wp__tlm_plan

Tool Management/ Tool Management Plan
-------------------------------------

This document implements parts of the :need:`wp__platform_mgmt`.

Purpose
+++++++
The purpose of the Tool Management Plan is to guide the identification, evaluation and qualification of
project tools.

> S-CORE objectives is to use OSS tools for the generation of all required work products. Thus all

Author: rephrased

Tool Management Goals
^^^^^^^^^^^^^^^^^^^^^

* Tools are identified and documented
* Tools are evaluated
* Tools are qualified, if applicable

Approach
++++++++

Identification
^^^^^^^^^^^^^^

Each tool is identified by its name or UID and version.
The tools are stored in one GitHub repository, `tooling repository <https://GitHub.com/eclipse-score/tooling/>`_.

For each tool the :need:`gd_temp__tool_management__verif_rpt_template` shall be used to document
the relevant information.

:need:`gd_chklst__tool__cr_review` shall be used to check that all relevant information is considered.

The configuration of tools is part of the :need:`doc__config_mgt_plan`.

Evaluation
^^^^^^^^^^

The tool confidence level (TCL) must be determined as outlined in the
:need:`doc_concept__tool__process`.

Step-by-Step guide:

1. Determine the tool impact
2. Determine the tool error detection
3. Determine the tool confidence level

**To Step 1**

Based on the found malfunctions or threats judge, if any safety or security requirements may violated.
If this is the case, determine tool impact to "YES" otherwise to "NO".

If tool impact is "NO", continue with 3, otherwise 2.

**To Step 2**

> Based on the found malfunctions or threats judge, if these can be all detected or prevented?

Reviewer 2: This is not a question and hard to understand. Better use:
```suggestion
Based on the malfunctions or threats found, assess whether these can be detected or prevented.
```

Author: taken over

Consider also potential safety measures and security controls to increase the detection and
prevention of these, if applicable.
If there is no guarantee to either detect or prevent these, judge tool error detection to "NO",
otherwise to "YES".

**To Step 3**

If tool impact is judged to "YES" and tool error detection judged to "NO", the tool confidence
level is determined to "LOW", otherwise "HIGH".

In case the tool confidence level is "LOW", tool qualification is required.

Qualification
^^^^^^^^^^^^^

> As method validation of software tool is applied.

Reviewer 2: The method used is the validation of the software tool.
Or 
The method used is software tool validation.
Or 
As method "validation of software tool" is applied.

Author: Modified to match last version

Specify requirements for the tools purpose and usage in the S-CORE context.
Analyze the malfunctions and threats and specify safety measures and security controls to increase
the prevention and detection of them.
Specify test cases and tests to verify the requirements and generate a test report.
Update the Tool Verification Report accordingly to document the outcomes.

For the generation of requirements, verification and analyses the processes defined in the

> :need:`wp__platform_mgmt` shall be used (Requirements Engineering, Verification,

Reviewer 2: Would it improve the quality to link to the respective parts for requirements, etc. rather than just say "read the platform management plan"?

Author: Add concrete links, as available for now

   # SPDX-License-Identifier: Apache-2.0
   # *******************************************************************************

.. document:: Tool Management Plan
   :id: doc__platform_tool_management_plan
   :status: draft
   :safety: ASIL_B
   :tags: platform_management
   :realizes: wp__tlm_plan

Tool Management/ Tool Management Plan
-------------------------------------

This document implements parts of the :need:`wp__platform_mgmt`.

Purpose
+++++++
The purpose of the Tool Management Plan is to guide the identification, evaluation and qualification of

> project tools.

Author: Please again, forgot looking only to safety, our process framework does conform to multiple standards and for you as user, only our process are relevant, the framework is mapped to the relevant standard and it defines, what we do for qualification, in our case. I add for current state just safe and secure use of tools, maybe replace in future by trustable use in the tool, where trustable may include more than safe and secure

> S-CORE objectives is to use OSS tools for the generation of all required work products. Thus all

Author: rephrased, all tools are under consideration, which may have impact on S-CORE product

tools which will be used as part of the tool chain for the work product generation must be identified,
documented and evaluated and if applicable, qualified.

Objectives and Scope
++++++++++++++++++++

Tool Management Goals
^^^^^^^^^^^^^^^^^^^^^

* Tools are identified and documented
* Tools are evaluated
* Tools are qualified, if applicable

Approach
++++++++

Identification
^^^^^^^^^^^^^^

> Each tool is identified by its name or UID and version.

Reviewer 3: Rephrase to
`Each tool is identified by its name or UID and its version.`

removes ambiguity

Author: changed as proposed

> The tools are stored in one GitHub repository, `tooling repository <https://GitHub.com/eclipse-score/tooling/>`_.

Author: rephrased as proposed

For each tool the :need:`gd_temp__tool_management__verif_rpt_template` shall be used to document
the relevant information.

:need:`gd_chklst__tool__cr_review` shall be used to check that all relevant information is considered.

The configuration of tools is part of the :need:`doc__config_mgt_plan`.

Evaluation
^^^^^^^^^^

The tool confidence level (TCL) must be determined as outlined in the
:need:`doc_concept__tool__process`.

> Step-by-Step guide:

Author: added the image for our process, just link to visualization of certain standard does not reflect, what we want

1. Determine the tool impact
2. Determine the tool error detection
3. Determine the tool confidence level

**To Step 1**

Based on the found malfunctions or threats judge, if any safety or security requirements may violated.
If this is the case, determine tool impact to "YES" otherwise to "NO".

If tool impact is "NO", continue with 3, otherwise 2.

**To Step 2**

Based on the malfunctions or threats found, assess whether these can be detected or prevented.
Consider also potential safety measures and security controls to increase the detection and

> prevention of these, if applicable.

Author: There maybe already some implemented from the tool provider, so that finally you judge it as High Confidence, but if they re not sufficient, you may need additional ones, then you create requirements for that and implement them and test it during qualification

If there is no guarantee to either detect or prevent these, judge tool error detection to "NO",
otherwise to "YES".

**To Step 3**

If tool impact is judged to "YES" and tool error detection judged to "NO", the tool confidence
level is determined to "LOW", otherwise "HIGH".

In case the tool confidence level is "LOW", tool qualification is required.

Qualification
^^^^^^^^^^^^^

> As method "validation of software tool" is applied.

Author: What development process? I just said, that we use S-CORE parts for tools too, and independent of that, having requirements or goals or how ever you call it, and you test that is minimum in my view

Specify requirements for the tools purpose and usage in the S-CORE context.

> Analyze the malfunctions and threats and specify safety measures and security controls to increase

Author: As part of the qualification, you may need to add more safety measures (could technical implementation or more process stuff) to increase the detection or prevention, so I added the word additional

the prevention and detection of them.
Specify test cases and tests to verify the requirements and generate a test report.

> Update the Tool Verification Report accordingly to document the outcomes.

Author: Add now overview with the state of the work products
